csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/TranslatedStmt.qll

```diff
@@ -916,3 +916,114 @@ class TranslatedLockStmt extends TranslatedStmt {
     result = LockElements::getLockWasTakenDecl(stmt)
   }
 }
+
+// TODO: Should be modeled using the desugaring framework for a
+//       more exact translation.
+class TranslatedCheckedUncheckedStmt extends TranslatedStmt {
+  TranslatedCheckedUncheckedStmt() {
+    stmt instanceof CheckedStmt or
+    stmt instanceof UncheckedStmt
+  }
+
+  override TranslatedElement getChild(int id) { id = 0 and result = this.getBody() }
+
+  override Instruction getFirstInstruction() { result = this.getBody().getFirstInstruction() }
+
+  override Instruction getChildSuccessor(TranslatedElement child) {
+    child = this.getBody() and
+    result = this.getParent().getChildSuccessor(this)
+  }
+
+  override predicate hasInstruction(
+    Opcode opcode, InstructionTag tag, Type resultType, boolean isLValue
+  ) {
+    none()
+  }
+
+  override Instruction getInstructionSuccessor(InstructionTag tag, EdgeKind kind) { none() }
+
+  private TranslatedElement getBody() {
+    result = getTranslatedStmt(stmt.(CheckedStmt).getBlock()) or
+    result = getTranslatedStmt(stmt.(UncheckedStmt).getBlock())
+  }
+}
+
+// TODO: Should be modeled using the desugaring framework for a
+//       more exact translation.
```

Comment:
I assume part of the future work is to generate a `finally` and to call the right `Dispose` method?